src/devices/machine/upd765.cpp

```diff
@@ -267,14 +270,18 @@ bool upd765_family_device::get_ready(int fid)
 	return !external_ready;
 }
 
-void upd765_family_device::set_ds(int state)
+void upd765_family_device::set_ds(int fid)
 {
-	for(int i = 0; i < 4; i++)
-	{
-		floppy_info &fi = flopi[i];
+	if (selected_drive == fid)
+		return;
+
+	// pass drive select to connected drives
+	for (auto &fi : flopi)
```

Comment:
Can you please not use `auto` like this?
* You definitely know what the type of object you want a reference to is
* The name is not excessively long or deeply qualified
* The type is not dependent of template arguments
* It is not an iterator or proxy to the actual object

Using `auto` here just makes the code less approachable.